src/basic/terminal-util.c

```diff
@@ -498,6 +498,15 @@ int release_terminal(void) {
         return r;
 }
 
+int terminal_new_session(void) {
+
+        if (!isatty_safe(STDOUT_FILENO) || !isatty_safe(STDERR_FILENO))
+                return -ENXIO;
+
+        (void) setsid();
+        return RET_NERRNO(ioctl(STDERR_FILENO, TIOCSCTTY, 0));
```

Comment:
so the ctty logic relevant for signal delivery mostly, i.e. if people hit C-c or the line is hung up. I'd claim such events are just a form of input, via a side channel (that side channel being signal delivery, i.e. SIGINT, SIGHUP et al). hence it would make a ton more sense to me if we'd use STDIN_FILENO here, i.e. focus on input, not output here.

(and hence also only check isatty_safe() for stdin above).

(and maybe add a brief comment explaining this reasoning)